internal/terraform/node_action_instance.go

```diff
@@ -60,7 +61,7 @@ func (n *NodeActionDeclarationInstance) Execute(ctx EvalContext, _ walkOperation
 	if err != nil {
 		diags = diags.Append(tfdiags.Sourceless(
 			tfdiags.Error,
-			"Failed to get provider",
+			fmt.Sprintf("Failed to get provider for %s", n.Addr),
```

Comment:
You could add a source range for the actual action here instead of putting the address in the summary?